wscript

```diff
@@ -78,41 +78,103 @@ def options(opt):
+    firewire = opt.add_auto_option(
+            'firewire',
+            help='Enable FireWire driver (FFADO)',
+            conf_dest='BUILD_DRIVER_FFADO')
+    firewire.check_cfg(
+            package='libffado >= 1.999.17',
+            args='--cflags --libs')
+    freebob = opt.add_auto_option(
+            'freebob',
+            help='Enable FreeBob driver')
+    freebob.check_cfg(
+            package='libfreebob >= 1.0.0',
+            args='--cflags --libs')
+    iio = opt.add_auto_option(
+            'iio',
+            help='Enable IIO driver',
+            conf_dest='BUILD_DRIVER_IIO')
+    iio.check_cfg(
+            package='gtkIOStream >= 1.4.0',
+            args='--cflags --libs')
+    iio.check_cfg(
+            package='eigen3 >= 3.1.2',
+            args='--cflags --libs')
+    portaudio = opt.add_auto_option(
+            'portaudio',
+            help='Enable Portaudio driver',
+            conf_dest='BUILD_DRIVER_PORTAUDIO')
+    portaudio.check(header_name='windows.h') # only build portaudio on windows
+    portaudio.check_cfg(
+            package='portaudio-2.0 >= 19',
+            uselib_store='PORTAUDIO',
+            args='--cflags --libs')
+    winmme = opt.add_auto_option(
+            'winmme',
+            help='Enable WinMME driver',
+            conf_dest='BUILD_DRIVER_WINMME')
+    winmme.check(
+            header_name=['windows.h', 'mmsystem.h'],
+            msg='Checking for header mmsystem.h')
+
+    celt = opt.add_auto_option(
+            'celt',
+            help='Build with CELT')
+    celt.add_function(check_for_celt)
+
+    # Suffix _PKG to not collide with HAVE_OPUS defined by the option.
+    opus = opt.add_auto_option(
+            'opus',
+            help='Build Opus netjack2')
+    opus.check(header_name='opus/opus_custom.h')
+    opus.check_cfg(
+            package='opus >= 0.9.0',
+            args='--cflags --libs',
+            define_name='HAVE_OPUS_PKG')
+
+    samplerate = opt.add_auto_option(
+            'samplerate',
+            help='Build with libsamplerate')
+    samplerate.check_cfg(
+            package='samplerate',
+            args='--cflags --libs')
+    sndfile = opt.add_auto_option(
+            'sndfile',
+            help='Build with libsndfile')
+    sndfile.check_cfg(
+            package='sndfile',
+            args='--cflags --libs')
+    readline = opt.add_auto_option(
+            'readline',
+            help='Build with readline')
+    readline.check(lib='readline')
+    readline.check(
+            header_name=['stdio.h', 'readline/readline.h'],
+            msg='Checking for header readline/readline.h')
+    sd = opt.add_auto_option(
+            'systemd',
+            help='Use systemd notify')
+    sd.check(header_name='systemd/sd-daemon.h')
+    sd.check(lib='systemd')
+    db = opt.add_auto_option(
+            'db',
+            help='Use Berkeley DB (metadata)')
+    db.check(header_name='db.h')
+    db.check(lib='db')
```

Comment:
Is the `db` dependency optional? Only use `add_auto_option` for dependencies that are optional.

Reply:
no waf expert here :) just pasted+copied from the opus and celt declarations before.

feel free to fix the wscript accordingly to standards (and for cross-platform too, as it probably only works for Linux as is).

thanks && cheers

Reply:
Okay, no problems; waf is kinda special. ;) Since there are not `#if HAVE_DB` checks in the code I think the `db` dependency should be mandatory. Please replace the snippet added to the `options` function with

```python
    conf.check(header_name='db.h')
    conf.check(lib='db')
```

somewhere in the `configure` function. Preferably after the check for `aften` for Mac OSX and before
```python
    conf.load('autooptions')
```

Reply:
maybe it could be optional after all.. will add `#ifdef HAVE_DB ... #endif" where applicable...
bbl